0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+6. At a later time, Alice or Bob may instruct their settlement engines to close the payment channel and release the allotted funds back to their respective ledger accounts.
+   - Depending upon the payment channel implementation, there may be both cooperative and non-cooperative mechanisms for this, with varying time delays until the funds can be dispersed.
+   - Settlement engines using payment channels may also have functionality to watch the underlying ledger to ensure the peer doesn't try to submit an old message that would incorrectly distribute the escrowed funds.
+
+### Sender collects a fee
+
+Suppose Alice and Bob's settlement arrangement is such that, in order the Alice to settle her liabilities to Bob, Alice requires that first Bob must pay a fee.
+
+1. The settlement engine implementations that Alice and Bob are operating might negotiate the fee that Alice will collect (suppose a fee of 3 units).
+   - The configuration of the settlement engines could ensure the fee is reasonable, or Bob could instruct his settlement engine of some limit for the fee.
+2. After it's negotiated, Bob's settlement engine preemptively acknowledges it received an incoming settlement for 3 units.
+   - Note that even though no settlement on the shared ledger or system was received, Alice and Bob did mutually agree to reconcile a future liability: that is, the fee charged by Alice on Bob's behalf.
+3. As Alice and Bob transact with one another, the liabilities that Alice owes Bob accrue, and Alice's accounting system instructs her settlement engine to settle them.
+4. Alice's settlement engine, instead of sending a settlement on the shared settlement system, should track the amount owed to Bob, until it accrues to 3 units, the value of the agreed fee.
+5. After the value of the fee is accounted for through accrued settlements, if Alice's accounting system instructs her settlement engine to settle, it should send these additional settlements on the shared system or ledger.
+   - For example, after negotiating the fee, suppose Alice's accounting system tells her settlement engine to settle 2 units. Nothing should happen, since it's accruing owed balance to cover the fee of 3 units.
+   - If the settlement engine is later instructed to settle another 2 units, it should send a payment on the shared settlement system for 1 unit, since the other unit was taken out as apart of the fee.
+   - Any further instructions to settle should trigger payments on the shared system as normal.
+
+### Recipient collects a fee
+
+Suppose Alice and Bob's settlement arrangement is such that, before Bob can begin acknowledging incoming settlements from Alice, Bob must collect some initiation fee related to the settlement system.
+
+1. Alice sends a custom instruction to her settlement engine to settle the fee. Note that she does not use the instruction reserved for the accounting system.
```

Comment:
This could be automated (the SE could use defaults in an "autopilot" -esque mode), or it'd be part of the "open payment channel" UX if Alice is manually performing an operation (I'm thinking about this in the context of paychans).